### ISSUE TITLE
revision of Kubernetes manifests

### ISSUE THREAD

**Issue by grossmannmartin** [shopsys/shopsys#1277]
### What is happening

Kubernetes and Kustomize are evolving and some concepts introduced earlier are now recommended to do differently.

- Kustomize introduced some major changes in 2.0 version and later in 3.0, 3.1
  - https://github.com/kubernetes-sigs/kustomize/blob/master/docs/v2.0.0.md
  - https://github.com/kubernetes-sigs/kustomize/blob/master/docs/v3.0.0.md
  - https://github.com/kubernetes-sigs/kustomize/blob/master/docs/v3.1.0.md
  - Kustomization Path Constraints should be taken into account – with current file structure is impossible to build config in Kustomize >2.0 version
  - Deprecated apiVersions should be updated:
    ```diff
    -  apiVersion: extensions/v1beta1
    +  apiVersion: apps/v1
       kind: Deployment
    ```
  - some of our manifests don't specify `spec.selector`, which shouldn't be possible [according to the Kubernetes docs](https://kubernetes.io/docs/concepts/workloads/controllers/deployment/#selector).
  - https://github.com/shopsys/kubernetes-buildpack should be updated with new versions

### Expected result
Manifests are reviewed with the latest versions of Kubernetes / Kustomize and updated accordingly.

**Comment by grossmannmartin** [comment]
In shopsys/demoshop#84 we discover the configuration files from application (parameters.yml, domain_urls.yml) have to be also in kubernetes folder. This should be solved properly too. (see https://github.com/shopsys/demoshop/pull/84#discussion_r336185745)

**Comment by grossmannmartin** [comment]
closed by #2643
